src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java

```diff
@@ -379,4 +379,38 @@ public void testNoHeaderMissingErrorInCaseHeaderSizeEqualToFileSize() throws Exc
         };
         verify(checkConfig, getPath("InputRegexpHeaderMulti52.java"), expected);
     }
+
+    @Test
+    public void testCharsetProperty1() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(RegexpHeaderCheck.class);
+        checkConfig.addProperty("headerFile", getPath("InputRegexpHeader7.header"));
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        final String path = getPath("InputRegexpHeader4.java");
+        verify(checkConfig, path, expected);
+    }
+
+    @Test
+    public void testCharsetProperty2() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(RegexpHeaderCheck.class);
+        checkConfig.addProperty("charset", "US-ASCII");
+        checkConfig.addProperty("headerFile", getPath("InputRegexpHeader7.header"));
+        final String[] expected = {
+            // -@cs[RegexpSinglelineJava] need for testing
```

Comment:
Try to use `id`.
Please put comment on whole method.

Reply:
It will not work if I put comment id 
![image](https://github.com/checkstyle/checkstyle/assets/97679350/debd3d50-6d37-4403-bac3-1487b18285a7)

It will also not work if i put this line at top of method 
![image](https://github.com/checkstyle/checkstyle/assets/97679350/abb37ba2-bfe4-4b07-a662-e341e038268e)
and
![image](https://github.com/checkstyle/checkstyle/assets/97679350/1fc5d0d2-2859-4a1e-a45e-561e4f4aa785)